lib/common/xml.c

```diff
@@ -2099,8 +2091,10 @@ xml_calculate_significant_changes(xmlNode *old_xml, xmlNode *new_xml)
 void
 xml_calculate_changes(xmlNode *old_xml, xmlNode *new_xml)
 {
-    CRM_CHECK(pcmk__xe_is(old_xml, crm_element_name(new_xml)), return);
-    CRM_CHECK(pcmk__str_eq(ID(old_xml), ID(new_xml), pcmk__str_casei), return);
+    CRM_CHECK((old_xml != NULL) && (new_xml != NULL)
+              && pcmk__xe_is(old_xml, (const char *) new_xml->name)
+              && pcmk__str_eq(ID(old_xml), ID(new_xml), pcmk__str_none),
+              return);
```

Comment:
I just remembered this, and it might be handy here (and likely a lot more other places) - there's a nonnull gcc attribute that can be attached to function parameters that tells the compiler to insert checks.  I'm not sure what exactly it does, whether it just adds a warning at build time or whether it causes something to happen at runtime.  We'd have to experiment with it.

Reply:
I believe it's compile-time only, and not particularly helpful